src/Nevermind/Nevermind.JsonRpc/Module/ModuleProvider.cs

```diff
@@ -0,0 +1,39 @@
+﻿using System.Collections.Generic;
+using System.Linq;
+using Nevermind.JsonRpc.DataModel;
+
+namespace Nevermind.JsonRpc.Module
+{
+    public class ModuleProvider : IModuleProvider
+    {
+        private readonly IConfigurationProvider _configurationProvider;
+        private IEnumerable<ModuleInfo> _modules;
+
+        public ModuleProvider(IConfigurationProvider configurationProvider, INetModule netModule, IEthModule ethModule, IWeb3Module web3Module, IShhModule shhModule)
+        {
+            _configurationProvider = configurationProvider;
+            Initialize(netModule, ethModule, web3Module, shhModule);
+        }
+
+        public IEnumerable<ModuleInfo> GetEnabledModules()
+        {
+            return _modules.Where(x => _configurationProvider.EnabledModules.Contains(x.ModuleType)).ToArray();
```

Comment:
same here, let us not create a new array each time, I guess we can reuse an object for this